### PR TITLE
Fix backprop through BatchNorm in testmode

### PR DESCRIPTION
**Pull request by baggepinnen**
This bug only appeared while taking gradients of networks is in testmode (`training=false`)

### CODE REVIEW

**Comment by baggepinnen** [comment]
Any reason this is held up?

**Comment by MikeInnes** [comment]
What's the reason backprop currently fails? Should we make sure `squeeze` is differentiable rather than just replacing it here? Would be great to have a test too.

**Comment by baggepinnen** [comment]
Well
- the `dims`-keyword is not supplied to `sum`
- `squeeze` is deprecated in favor of `dropdims`

**Comment by MikeInnes** [comment]
Ah, I thought Flux had defined its own version of `squeeze` at some point, but I must be thinking of something else. Just needs the conflict fixed and the test then, thanks!

**Comment by CarloLucibello** [comment]
closing as outdated
